File: typical90/day07.py
n=int(input())
a=[*map(int,input().split())]
q=int(input())
a.sort()
m=max(a)
for _ in range(q):
    b=int(input())
    ok=-1
    ng=m
    while abs(ok-ng)>1:
        mid=(ok+ng)//2
        if mid>=b:
            ok=mid
        else:
            ng=mid
    print(min(abs(ok-b),abs(ng-b)))

File: rust/src/bin/day07.rs
use proconio::input;

fn main(){
    input!{
        n:usize,
        mut a:[usize;n],
        q:usize,
        b:[usize;q]
    };
    a.sort();
    for i in b{
        let mut ok = 0;
        let mut ng = n-1;
        while ((ok-ng) as isize).abs()>1 {
            let mid:usize=(ok+ng)/2;
            println!("mid: {}",mid);
            if a[mid]<i {
                ok=mid;
            }else{
                ng=mid;
            }
        }
        println!("{}",ok)
    }
}